app/services/query-service.js

```diff
@@ -47,25 +52,27 @@ const validateResponseLinks = (response, query) => {
       return trackIssueAndBreak('validateResponseLinks failed because hallucinated links detected in response objects')
     }
 
-    const invalidLinks = responseEntriesAndLinks.filter((entry) =>
-      entry.matches.some((link) => !trueEntriesAndLinks.some((trueEntry) => trueEntry.matches.includes(link)))
-    )
+    const trueMatches = trueEntriesAndLinks.map(entry => entry.matches).flat()
+    const responseMatches = responseEntriesAndLinks.map(entry => entry.matches).flat()
+
+    const invalidLinks = responseMatches.filter(link => !trueMatches.some(trueLink => trueLink === link || trueLink.includes(link)))
 
     if (invalidLinks !== undefined && invalidLinks.length > 0) {
       return trackIssueAndBreak('validateResponseLinks failed because invalid links detected in response objects')
     }
-  } catch {
+  } catch (e) {
+    logger.error(e, 'validateResponseLinks failed because of an error')
```

Comment:
Can change this message and the trackIssueAndBreak func to "validateResponseLinks failed"